### PR TITLE
adds in workspace header to control CTAs / workspace view controls

### LINKED ISSUE #1113: Create a new header component that also contains a workspaces CTA

### PR DESCRIPTION
**Pull request by hamilton**
closes #1113 

What's left:
- [x] finesse metrics / dashboard
- [x] finesse spacing in inspector
- [x] finesse splash page
- [x] add tooltips to new workspace header buttons
- [x] fix any lint / prettier errors
- [x] fix nav button spacing issues

### CODE REVIEW

**Comment by magorlick** [comment]
Thank you for these changes!  Overall I think this looks great, but I found a few nits we might improve upon before merging: 

P1) CTA buttons are missing the dynamic resizing to just the icon for small screen widths for very narrow widths.

image

P1) Why are we treating the pane collapse button differently than the right? it is really nice to be able to click in place to toggle the panes. Consider adding this button to the header menu too to the left of the other pane buttons.

image

P1) The CTA tool tips feel a little wordy and misaligned across the application. Some suggestions:

**Sources**:
image
"Model the data source with SQL"
image
"Dashboard the data source with automated metrics"


**Models**:
image
"Export the modeled data as file"

image
"Dashboard the data model with automated metrics"


**Metrics**:
image
"Explore the metrics dashboard"


**Dashboard**:
image
"Edit the metrics that power this dashboard"

**Comment by bcolloran** [comment]
I agree with all of Marissa's comments, in particular being able to toggle the left hand nav pane with a button that remains in place seems and convenient. In both notion and slack for example, the collapse/expand button always moving around is super annoying -- I bet we can do better than those.  Seeing this now I wonder if the nav header bar ought to go all the way across the top of the viewport, in which case that the left hand nav pane toggle button could live in the header bar, in the same level of the hierarchy as the other pane toggles. I wouldn't block on that larger rethink for getting this PR in, however I do have a couple other comments that I think we might want to try to address before merging this one and, and we can go back and think about the nav placement more holistically in another PR, because this is an overall improvement for sure


Seeing this live, I think that the header bar in the dashboard looks a little off relative to the header and the other surfaces -- the thin gray line that separates the top area from the dashboard area almost makes it look like the head the header bar has expanded down or something. I think we could just fade out that thin grained line in the dashboard view, we really don't need it at all
image

Also noticing that the edit metrics button gets pushed pushed off the screen even at pretty wide screen widths:
image

The icon for toggling the bottom pane looks a bit heavy (dark overall, a lot of black ink) and vibratey (thin white line next to thicker black line)-- I wonder  whether we could try some other concepts for that, though this is really not a big deal
image


Hamilton, you also asked me in specific to check out the default widths of the side panes -- I think the left hand navigation pane default width is fine, but it seems to be equal to the minimum width, no? I think that this width is fine as a default, but I think it should be possible to shrink the pane a bit smaller than that default, so I might reduce the minimum, so I might reduce the minimum allowed width a bit. No comments on the inspector pane

**Comment by hamilton** [comment]
Thanks for the feedback @magorlick and @bcolloran. A on a few topics:
- re: the tooltips ~ @magorlick happy to change these. I think these were in the application already, so this is a good occasion to tune them up for 18.
- re: the buttons + nav, I don't think these should actually go together. The navigation is higher-up in the hierarchy and not every workspace / view has these types of collapse buttons (see the onboarding for example, or the current dashboard). Open to changing these later once we have a better idea of how the hierarchy might change; our mental model might change down the line.

File: web-local/src/lib/components/workspace/core/WorkspaceHeader.svelte
Comment (ericpgreen2) [inline web-local/src/lib/components/workspace/core/WorkspaceHeader.svelte:65]: Delete?

File: web-local/src/lib/components/workspace/metrics-def/MetricsDefWorkspaceHeader.svelte
Comment (ericpgreen2) [inline web-local/src/lib/components/workspace/metrics-def/MetricsDefWorkspaceHeader.svelte:24]: Leftover comment

File: web-local/src/lib/components/workspace/metrics-def/MetricsDefWorkspaceHeader.svelte
Comment (ericpgreen2) [inline web-local/src/lib/components/workspace/metrics-def/MetricsDefWorkspaceHeader.svelte:30]: Leftover comment

File: web-local/src/lib/components/workspace/model/ModelBody.svelte
Comment (ericpgreen2) [inline web-local/src/lib/components/workspace/model/ModelBody.svelte:61]: Delete this commented-out code?

File: web-local/src/lib/components/workspace/model/ModelWorkspaceHeader.svelte
Comment (ericpgreen2) [inline web-local/src/lib/components/workspace/model/ModelWorkspaceHeader.svelte:38]: Leftover comment

**Review by ericpgreen2 — APPROVED** [review]
Pointed out a few places where it looks like old code was commented out rather than deleted. Otherwise looks good!

**Pull request merged**
